hazelcast/src/main/java/com/hazelcast/config/OnJoinPermissionOperationName.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+/**
+ * Enum of operation names for handling client permissions when the member is joining into the cluster.
+ *
+ * @see SecurityConfig
+ */
+public enum OnJoinPermissionOperationName {
```

Comment:
Minor: usually these are suffixed by `Type`

Reply:
I will keep it unchanged for now. When we decide to change it we can do it in a next PR.

We already have this name use in the Hazelcast release 3.11.2 .